### PR TITLE
AWS EKS How-to-guides Document

### PR DESCRIPTION
**Pull request by garvit3835**

### CODE REVIEW

**Comment by vikasgoswami2** [comment]
@mishushakov please review this

File: how-to-guides/cloud-services/aws/connect-to-an-eks-cluster.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+{% code overflow="wrap" lineNumbers="false" %}
+```bash
+aws configure
+```
+{% endcode %}
+
+3. Check if the cluster is accessible:
+
+{% code overflow="wrap" lineNumbers="false" %}
```
Comment (dray92) [inline how-to-guides/cloud-services/aws/connect-to-an-eks-cluster.md:104]:
```suggestion
{% code overflow="wrap" %}
```

File: how-to-guides/cloud-services/aws/connect-to-an-eks-cluster.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+```bash
+kubectl get svc
+```
+
```
Comment (dray92) [inline how-to-guides/cloud-services/aws/connect-to-an-eks-cluster.md:108]:
```suggestion
{% code overflow="wrap" %}
```bash
kubectl get svc
```
{% endcode %}
```

**Review by dray92 — APPROVED** [review]

File: how-to-guides/cloud-services/aws/connect-to-an-eks-cluster.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+## Prerequisites:
+
+- Follow the [Connecting to AWS](../../existing-network/connecting-to-aws.md) guide.
+
+## Existing EKS
+
+### Step 1: Accessing EKS from a DevBox
+
+1. Go to **Amazon Elastic Kubernetes Service > Clusters**.
+2. Go to the **DevBox** and install awscli:
```
Comment (mishushakov) [inline how-to-guides/cloud-services/aws/connect-to-an-eks-cluster.md:17]: your

File: how-to-guides/cloud-services/aws/connect-to-an-eks-cluster.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
+# EKS
+Connecting to EKS running in the private subnet of AWS VPC from your DevBox.
+
+## Architecture Diagram:
+
+![image](../../../.gitbook/assets/eks-architecture.png)
+
+## Prerequisites:
+
+- Follow the [Connecting to AWS](../../existing-network/connecting-to-aws.md) guide.
```
Comment (mishushakov) [inline how-to-guides/cloud-services/aws/connect-to-an-eks-cluster.md:10]: remove the

File: how-to-guides/cloud-services/aws/connect-to-an-eks-cluster.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+3. Download the **kubectl** binaries:
+
+{% code overflow="wrap" lineNumbers="false" %}
+```bash
+curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
+```
+{% endcode %}
+
+4. Install the **kubectl**:
```
Comment (mishushakov) [inline how-to-guides/cloud-services/aws/connect-to-an-eks-cluster.md:33]: remove the

File: how-to-guides/cloud-services/aws/connect-to-an-eks-cluster.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+kubectl get svc
+```
+{% endcode %}
+
+![image](../../../.gitbook/assets/eks-access.png)
+
+
+## New EKS Cluster
+
+### Step 1: Creating an EKS Cluster
```
Comment (mishushakov) [inline how-to-guides/cloud-services/aws/connect-to-an-eks-cluster.md:54]: creating a

File: how-to-guides/cloud-services/aws/connect-to-an-eks-cluster.md [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+
+1. Go to **Amazon Elastic Kubernetes Service > Clusters > Add cluster > Create**.
+2. Enter name of the cluster and assign a **Cluster service role** that has **AmazonEKSClusterPolicy** permission. If no role is present, create a new one with the same permission.
```
Comment (mishushakov) [inline how-to-guides/cloud-services/aws/connect-to-an-eks-cluster.md:57]: the name, which has

File: how-to-guides/cloud-services/aws/connect-to-an-eks-cluster.md [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+3. Click on the **Next**. In the **Networking** section, choose the VPC where your bastion host is running on and assign 2 private subnets . In the **Cluster endpoint access**, select **Private**.
```
Comment (mishushakov) [inline how-to-guides/cloud-services/aws/connect-to-an-eks-cluster.md:58]: Click on next, running on > deployed. Remove space before dot. In the cluster endpoint access > in cluster endpoint access

File: how-to-guides/cloud-services/aws/connect-to-an-eks-cluster.md [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+4. Configure other configurations as required and then create the cluster. After creating the cluster, go to the **compute** section and under **Node groups** click on **Add node group**.
+5. Enter the name of the node group and assign a node IAM role with **AmazonEC2ContainerRegistryReadOnly**, **AmazonEKS_CNI_Policy** and **AmazonEKSWorkerNodePolicy** permission. If no role is present, create a new one with the same permissions.
```
Comment (mishushakov) [inline how-to-guides/cloud-services/aws/connect-to-an-eks-cluster.md:60]: permissions

File: how-to-guides/cloud-services/aws/connect-to-an-eks-cluster.md [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+6. Configure other configurations as required and create the nodes.
+
+![imgae](../../../.gitbook/assets/eks-cluster.png)
+
+### Step 2: Installing necessary tools:
```
Comment (mishushakov) [inline how-to-guides/cloud-services/aws/connect-to-an-eks-cluster.md:65]: the necessary tools

File: how-to-guides/cloud-services/aws/connect-to-an-eks-cluster.md [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+
+1. Go to the **DevBox** and install awscli:
```
Comment (mishushakov) [inline how-to-guides/cloud-services/aws/connect-to-an-eks-cluster.md:67]: your DevBox

File: how-to-guides/cloud-services/aws/connect-to-an-eks-cluster.md [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+
+{% code overflow="wrap" lineNumbers="false" %}
+```bash
+sudo apt install awscli
+```
+{% endcode %}
+
+2. Download the **kubectl** binaries:
```
Comment (mishushakov) [inline how-to-guides/cloud-services/aws/connect-to-an-eks-cluster.md:75]: Download kubectl binaries

File: how-to-guides/cloud-services/aws/connect-to-an-eks-cluster.md [hunk 13]
```diff
@@ -0,0 +1,108 @@
...
+
+{% code overflow="wrap" lineNumbers="false" %}
+```bash
+curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
+```
+{% endcode %}
+
+3. Install the **kubectl**:
```
Comment (mishushakov) [inline how-to-guides/cloud-services/aws/connect-to-an-eks-cluster.md:83]: Install kubectl

**Review by mishushakov — COMMENTED** [review]
minor grammar

**Comment by vikasgoswami2** [comment]
@mishushakov done with the changes. Pls check now

**Pull request merged**